### PR TITLE
Move utilities + add some tests

### PR DESCRIPTION
**Pull request by aferr**
Fixes https://github.com/google-research/raksha/issues/346

### CODE REVIEW

File: src/utils/move_append.h [hunk 1]
```diff
@@ -16,16 +16,18 @@
...
+// the `dst` vector.
 template <typename T>
 void MoveAppend(std::vector<T>& dst, std::vector<T>&& src) {
   dst.insert(dst.end(), std::make_move_iterator(src.begin()),
              std::make_move_iterator(src.end()));
 }
 
-}  // namespace raksha::ir::auth_logic
-#endif  // SRC_IR_AUTH_LOGIC_MOVE_APPEND_H_
+}  // namespace raksha::utils
```
Comment (bgogul) [inline src/utils/move_append.h:32]: newline after this

File: src/utils/map_iter_test.cc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#include "src/utils/map_iter.h"
+
+#include "src/common/testing/gtest.h"
+
+namespace raksha::utils {
+TEST(MapIterTestSuite, SimpleMapIterTest) {
+  std::vector test_vec = MapIter<int, int>(std::vector({1, 5, 3, 9, 5}),
+                                           [](const int& x) { return x + 3; });
+  EXPECT_EQ(test_vec, std::vector({4, 8, 6, 12, 8}));
+}
```
Comment (bgogul) [inline src/utils/map_iter_test.cc:25]: newline after `}`

File: src/utils/move_append_test.cc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+#include "src/common/testing/gtest.h"
+
+namespace raksha::utils {
+TEST(MoveAppendTestSuite, SimpleMoveAppendTest) {
+  std::vector test_vec1({1, 2, 3, 4});
+  std::vector test_vec2({5, 6, 7, 8});
+  MoveAppend(test_vec1, std::move(test_vec2));
+  EXPECT_EQ(test_vec1, std::vector({1, 2, 3, 4, 5, 6, 7, 8}));
+}
```
Comment (bgogul) [inline src/utils/move_append_test.cc:29]: newline after `}`

File: src/utils/move_append_test.cc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+//-----------------------------------------------------------------------------
+
+#include "src/utils/move_append.h"
+
+#include <vector>
+
+#include "src/common/testing/gtest.h"
+
+namespace raksha::utils {
+TEST(MoveAppendTestSuite, SimpleMoveAppendTest) {
```
Comment (bgogul) [inline src/utils/move_append_test.cc:24]: newline between lines  23 and 24.

File: src/utils/map_iter_test.cc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#include "src/utils/map_iter.h"
+
+#include "src/common/testing/gtest.h"
+
+namespace raksha::utils {
+TEST(MapIterTestSuite, SimpleMapIterTest) {
```
Comment (bgogul) [inline src/utils/map_iter_test.cc:21]: newline between lines 20 and 21

**Review by bgogul — APPROVED** [review]
Just some minor nits.

File: src/utils/map_iter_test.cc (hunk 5 above)
Comment (aferr) [reply src/utils/map_iter_test.cc:22]: hmmm... my linter didn't catch this or require this. similar with other newlines.

**Pull request merged**